sdk/go/pulumi/types.go

```diff
@@ -687,16 +687,21 @@ func awaitInputs(ctx context.Context, v, resolved reflect.Value) (bool, bool, []
 		// but `T` does (https://golang.org/ref/spec#Method_sets).
 		// In this case, dereference the pointer to get at its actual value.
 		if v.Kind() == reflect.Ptr && valueType.Kind() != reflect.Ptr {
-			v = v.Elem()
-			contract.Assert(v.Interface().(Input).ElementType() == valueType)
+			if v.Elem().CanInterface() {
```

Comment:
This change essentially copies the Assert into the if condition. The test made assert fail here, though it is not the relevant case, not what the comment describes.